lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -205,6 +266,74 @@ export default function SubmitGradeForm({
             )}
           </span>
 
+          {acceptGradingComments && (
+            <span className="relative">
+              <Button
+                icon={commentIsSet ? NoteFilledIcon : NoteIcon}
+                disabled={disabled}
+                title={commentIsSet ? 'Edit comment' : 'Add comment'}
+                onClick={() => setShowCommentControls(prev => !prev)}
+                classes={classnames(
+                  'border border-r-0 rounded-none ring-inset h-full relative',
+                  'disabled:opacity-50'
+                )}
+              />
+              {showCommentControls && (
+                <div
+                  className={classnames(
+                    'w-80 p-3',
+                    'shadow border rounded bg-white',
+                    'absolute top-[calc(100%+3px)] right-0'
+                  )}
+                >
```

Comment:
3 and 4 should also be easy, as we already have hooks for that in frontend-shared.

If anything, closing when clicking away might be tricky because we are dealing with multiple frames here, but at least closing when clicking away inside the same frame should be straightforward.

Reply:
> If anything, closing when clicking away might be tricky because we are dealing with multiple frames here, but at least closing when clicking away inside the same frame should be straightforward.

Yes, I can't remember what events the parent frame will get when the iframe gains focus and whether the hooks we have will respond to those. For the first step, making same-frame clicks close the popover will be fine.